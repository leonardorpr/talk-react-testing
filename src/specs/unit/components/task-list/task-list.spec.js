import React from 'react'
import { render } from 'react-testing-library'
import 'jest-styled-components'

import { mockTasks } from 'specs/helpers'

import TaskList, { renderTasks } from 'components/task-list'

describe('[COMPONENTS] TaskList', () => {
  it('Should render correctly', () => {
    const { container } = render(<TaskList tasks={mockTasks} title='Done' toggle={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  it('Should return correct title', () => {
    const { container } = render(<TaskList tasks={mockTasks} title='Done' toggle={jest.fn()} />)
    expect(container.querySelector('h3').textContent).toEqual('Done')
  })

  it('Should list all tasks', () => {
    const tasks = renderTasks(mockTasks, jest.fn())
    expect(tasks).toHaveLength(3)
  })
})
