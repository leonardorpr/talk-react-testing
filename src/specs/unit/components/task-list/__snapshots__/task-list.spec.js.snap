// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] TaskList Should render correctly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  max-width: 400px;
  height: 60px;
  margin-left: 20px;
  cursor: pointer;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  margin: 15px;
}

.c3:hover {
  -webkit-box-shadow: 0px 2px 17px -5px rgba(0,0,0,0.75);
  -moz-box-shadow: 0px 2px 17px -5px rgba(0,0,0,0.75);
  box-shadow: 0px 2px 17px -5px rgba(0,0,0,0.75);
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c5 {
  background-color: #2A2C30;
  width: 97.5%;
  border-top-right-radius: 6px;
  border-bottom-right-radius: 6px;
  padding: 10px;
}

.c4 {
  background-color: #2ECC71;
  width: 2.5%;
  margin-right: 2px;
  border-top-left-radius: 6px;
  border-bottom-left-radius: 6px;
}

.c9 {
  background-color: #DA493A;
  width: 2.5%;
  margin-right: 2px;
  border-top-left-radius: 6px;
  border-bottom-left-radius: 6px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  color: #FFF;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c8 {
  text-align: end;
  font-size: 12px;
  color: #FFF;
}

.c0 {
  width: 100%;
  max-width: 900px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  color: #FFF;
  font-size: 20px;
  text-transform: uppercase;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h3
        class="c2"
      >
        Done
      </h3>
      <div
        class="c3"
        id="card"
      >
        <div
          class="c4"
          id="indicator"
        />
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <span
              class="c7"
            >
              Make Coffee
            </span>
            <span
              class="c8"
            >
              15 minutes
            </span>
          </div>
        </div>
      </div>
      <div
        class="c3"
        id="card"
      >
        <div
          class="c9"
          id="indicator"
        />
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <span
              class="c7"
            >
              Study
            </span>
            <span
              class="c8"
            >
              3 hours
            </span>
          </div>
        </div>
      </div>
      <div
        class="c3"
        id="card"
      >
        <div
          class="c9"
          id="indicator"
        />
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <span
              class="c7"
            >
              Dinner
            </span>
            <span
              class="c8"
            >
              1 hours
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
